src/core/job.c

```diff
@@ -827,11 +827,11 @@ static void job_fail_dependencies(Unit *u, UnitDependency d) {
                 if (!IN_SET(j->type, JOB_START, JOB_VERIFY_ACTIVE))
                         continue;
 
-                job_finish_and_invalidate(j, JOB_DEPENDENCY, true);
+                job_finish_and_invalidate(j, JOB_DEPENDENCY, true, false);
         }
 }
 
-int job_finish_and_invalidate(Job *j, JobResult result, bool recursive) {
+int job_finish_and_invalidate(Job *j, JobResult result, bool recursive, bool already) {
```

Comment:
I figure, if we didn't have to care for API we would simply fold this into a new JobResult value JOB_DONE_ALREADY or so, but I figure we can't do that now, given that this is propagated onto the bus.

I don't really like functions with lots of bool parameters, but I don#t have a better idea here, and defining a flags enum here, appears like overkill, too, so I guess this is OK.